packages/server/graphql/types/Task.ts

```diff
@@ -181,6 +182,28 @@ const Task: GraphQLObjectType = new GraphQLObjectType<any, GQLContext>({
             sendToSentry(error, {userId: accessUserId})
           }
           return data
+        } else if (integration.service === 'gitlab') {
+          const gitlabAuth = await dataLoader
+            .get('teamMemberIntegrationAuths')
+            .load({service: 'gitlab', teamId, userId: viewerId})
+          const accessToken = gitlabAuth?.accessToken
+          if (!accessToken) return null
+          const {gid} = integration
+          const query = `
+          {
+             issue(id: "${gid}"){
+                   __typename
+                   id
+                   iid
+                   title
```

Comment:
When we query `xGitLabIssue` on the frontend, we use the GitLab schema which says that several fields, such as `author.id`, are non-nullable. However, these are nullable fields for us as we're not returning them from this `Task` query. 

There are over a dozen non-nullable fields for GitLab issues. If we want to return all of them, the query would be very expensive. We'd need to do this in the `UpdatePokerScopeMutation` optimistic updater too. This also applies to our GitHub logic.

I think it's the right approach to only return the fields that we're likely to use, but nullable fields disguised as non-nullable fields certainly fields error prone. I know that we're refactoring the way that task integrations work, so this might be worth considering.